spec/steps/provision.fmf

```diff
@@ -97,3 +96,47 @@ example: |
             how: connect
             guest: hostname or ip address
             key: private-key-path
+
+/environment:
+    summary: Environment specification
+
+    description: |
+        As part of the ``provision`` step it is possible to
+        specify additional requirements for the testing
+        environment such as required amount of ``memory`` and
+        ``disk`` space.
+
+        FIXME: Much more details go here. For now let's start with
+        a couple of tangible examples.
+
+    example: |
```

Comment:
we should mention by default different properties are considered to all match, i.e. "and"-ed

Reply:
Right, but then, it's up to provisioner's behaviour to decide the details, errors etc..

There's no "generic" matcher. So the logic is not enforced anywhere, right?

Reply:
Added a paragraph explicitly documenting this.